napi-inl.h

```diff
@@ -4083,34 +4070,34 @@ inline napi_status ThreadSafeFunction::NonBlockingCall(
 
 inline void ThreadSafeFunction::Ref(napi_env env) const {
   if (_tsfn != nullptr) {
-    napi_status status = napi_ref_threadsafe_function(env, *_tsfn);
+    napi_status status = napi_ref_threadsafe_function(env, _tsfn);
     NAPI_THROW_IF_FAILED_VOID(env, status);
   }
 }
 
 inline void ThreadSafeFunction::Unref(napi_env env) const {
   if (_tsfn != nullptr) {
-    napi_status status = napi_unref_threadsafe_function(env, *_tsfn);
+    napi_status status = napi_unref_threadsafe_function(env, _tsfn);
     NAPI_THROW_IF_FAILED_VOID(env, status);
   }
 }
 
 inline napi_status ThreadSafeFunction::Acquire() const {
-  return napi_acquire_threadsafe_function(*_tsfn);
+  return napi_acquire_threadsafe_function(_tsfn);
 }
 
 inline napi_status ThreadSafeFunction::Release() {
-  return napi_release_threadsafe_function(*_tsfn, napi_tsfn_release);
+  return napi_release_threadsafe_function(_tsfn, napi_tsfn_release);
 }
 
 inline napi_status ThreadSafeFunction::Abort() {
-  return napi_release_threadsafe_function(*_tsfn, napi_tsfn_abort);
+  return napi_release_threadsafe_function(_tsfn, napi_tsfn_abort);
 }
 
 inline ThreadSafeFunction::ConvertibleContext
 ThreadSafeFunction::GetContext() const {
   void* context;
-  napi_get_threadsafe_function_context(*_tsfn, &context);
+  napi_get_threadsafe_function_context(_tsfn, &context);
```

Comment:
Abort if failed here?

Reply:
Since the change doesn't change the behavior here, we could address this issue in another PR. Opened issue here https://github.com/nodejs/node-addon-api/issues/581.

Reply:
@legendecas I agree, can you open an issue to make sure we fix it later?

Reply:
@mhdawson the issue is above, and PR for fix (which is also conflicting with this PR) is #583

Reply:
@KevinEady - oops should have read more carefullly. Thanks.